### ISSUE TITLE
after updating magento 2.0.4 from Magento 2 login popup is not working

### ISSUE THREAD

**Issue by inborntech** [magento/magento2#4220]
Site url is www.dilsedeal.com

**Comment by vrann** [comment]
@inborntech is it Customer login on the front page you are talking about? Is it customized on your site? Can you please check is there any errors in the logs

**Comment by inborntech** [comment]
@vrann 

Screenshot link shared please check.

Magento 2.0.4 : http://prnt.sc/atybzi

Magento 2.0.2 : http://prnt.sc/atycr1

No errors log present.

**Comment by mazhalai** [comment]
@inborntech did you regenerate static content? `bin/magento setup:static-content:deploy`

**Comment by inborntech** [comment]
@mazhalai  same no changes after regenerate static content.

**Comment by mazhalai** [comment]
@inborntech any errors on browser's console?

**Comment by inborntech** [comment]
@mazhalai  Tried Google Chrome, Internet Explorer, Microsoft Edge, Mozila Firefox same issue..

All popups are working only "Login Popup" is not working.

**Comment by mazhalai** [comment]
@inborntech sorry we are not able to reproduce this error.

**Comment by inborntech** [comment]
@mazhalai Kindly visit this site www.dilsedeal.con can you give me any clue how to enable popup?

**Comment by NanthiniMuralidaran** [comment]
@mazhalai In this case,I faced that popup issue.
I added a product to the cart before log in to the site.Then i logged out from the site.Again i added a product to the cart before log in.And then i logged in and clicked "Proceed to checkout",the popup appears.

**Comment by piotrekkaminski** [comment]
Anyone else experiencing this issue?

**Comment by LS-Supriya** [comment]
Even I am facing this Issue.

**Steps to Reproduce:**
1. Add a product to cart
2.Click on Proceed to Checkout
3.Page will be redirected to Login page (as guest checkout is disabled - handled in code)
4. Now Again add a product to cart
5. Login to Magento
6. Click on Proceed to checkout (in our case page will be automatically redirected to Edit cart page when items are present in the cart during login)
7. Pop up appears and prompting users to login instead of redirecting to check out page[
![checkout](https://cloud.githubusercontent.com/assets/17175033/15498823/bdcf159a-21be-11e6-9a55-f63c0dab5a99.png)
](url)

**Comment by mazhalai** [comment]
@inborntech @LS-Supriya do you have any custom modules?

**Comment by LS-Supriya** [comment]
Yes. We are developing our own module

**Comment by OwenRay** [comment]
I'm facing the same issue,
I'm using magento 2.0.7

What I do:
- opened the website in an incognito tab
- logged in
- added an item to my cart
- clicked checkout
- the popup appears
  note: I am still logged in, I can access my account settings and the logout button is displayed on the top right

I've tried disabling all 3rd party plugins
I've tried switching to the default Magento Luma Theme.
These did not help.

**Comment by ricardojle** [comment]
I have the same problem.
The best way to simulate the problem:
- change the config -> Settings / Sales / Checkout / After Adding a Product Redirect to Shopping Cart = yes
- delete the browser cache
- add item to my cart
- click proceed to checkout

The problem occurs only once , that is, on the first visit . Then it works normally.

**Comment by qubaish-techverx** [comment]
Did any body found solution of this issue ? @ricardojle i repeat the points which you suggest but it did not worked for me .

**Comment by ludwig-gramberg** [comment]
we had the same problem on our 2.0.7 installation due to a custom implementation, maybe it will help someone:

problem was this:
we implemented a redirect to the account creation page using a controller which would set a session-variable (to "flag" this account creation as something special) the users would then be put into a different user group right away.
problem was we stored our flag inside the customer session and that triggered exactly this problem with the login popup (also the welcome message would not appear at first).
we then fixed it by using another session model just for our module and now it works as intended and the bug is gone.

tl;dr
check if you manipulate the customer session anywhere in your code, if so, use a custom session model, i.e:
```
<?php
namespace Your\Namespace\Here;

class Session extends \Magento\Framework\Session\SessionManager {

}
```

**Comment by magento-engcom-team** [comment]
@inborntech, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
We tested the issue on 2.3.0-dev, 2.2.0, 2.1.9
